
<launch>
    <arg name="scene_fn" default="/home/rudorfem/ros/example_scene_A2.yaml" doc="Path to the Scene.yaml file"/>
    <arg name="vis" default="true" doc="Launch RViZ"/>
    <arg name="fps_limit" default="15" doc="frame rate limit (to save computation power)"/>
    <arg name="camera" default="realsense" doc="camera to use [kinect2, realsense]"/>
    <arg name="kinect_quality" default="hd" doc="kinect2 quality [sd, qhd, hd]"/>

    <!-- setting up kinect2 sensor -->
    <include file="$(find kinect2_bridge)/launch/kinect2_bridge.launch" if="$(eval camera=='kinect2')">
        <arg name="fps_limit" value="$(arg fps_limit)"/>
        <arg name="publish_tf" value="true"/>
    </include>

    <!-- setting up realsense sensor /camera/depth/color/points -->
    <include file="$(find realsense2_camera)/launch/rs_camera.launch" if="$(eval camera=='realsense')">
        <arg name="depth_width"       value="1280"/>
        <arg name="depth_height"      value="720"/>
        <arg name="depth_fps"         value="$(arg fps_limit)"/>
        <arg name="color_width"       value="1280"/>
        <arg name="color_height"      value="720"/>
        <arg name="color_fps"         value="$(arg fps_limit)"/>
        <arg name="enable_depth"      value="true"/>
        <arg name="enable_color"      value="true"/>
        <arg name="enable_infra1"     value="false"/>
        <arg name="enable_infra2"     value="false"/>
        <arg name="enable_fisheye"    value="false"/>
        <arg name="enable_gyro"       value="false"/>
        <arg name="enable_accel"      value="false"/>
        <arg name="enable_sync"       value="true"/>
        <arg name="filters"           value="decimation,temporal,pointcloud"/>
    </include>

    <!-- start up scene visualizer -->
    <node pkg="burg_scene_visualizer" type="visualize_scene.py" name="visualize_scene" output="screen">
        <!-- kinect2 settings - is the camera frame correct? -->
        <remap from="~input/image" to="/kinect2/$(arg kinect_quality)/image" if="$(eval camera=='kinect2')" />
        <remap from="~input/camera_info" to="/kinect2/$(arg kinect_quality)/image" if="$(eval camera=='kinect2')" />
        <param name="camera_frame" value="kinect2_link" if="$(eval camera=='kinect2')" />

        <!-- realsense settings -->
        <remap from="~input/image" to="/camera/color/image_raw" if="$(eval camera=='realsense')" />
        <remap from="~input/camera_info" to="/camera/color/camera_info" if="$(eval camera=='realsense')" />
        <param name="camera_frame" value="camera_color_optical_frame" if="$(eval camera=='realsense')" />

        <param name="scene_fn" value="$(arg scene_fn)" />
    </node>

    <!-- show all output data in rviz -->
    <node pkg="rviz" type="rviz" respawn="false" name="rviz"
          args="-d $(find burg_scene_visualizer)/rviz/visualizer-config.rviz" if="$(arg vis)"/>

</launch>
